### PR TITLE
Organizing test settings files by category

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Organized the settings in the test settings files by category.

> Take a look and see what you think. I generated these files via scripting, so I can make tweaks if need be.

## Why is the change being made?

@albeanth [recently suggested](https://github.com/terrapower/armi/discussions/1083) that ARMI settings files could have categories to help us organize them. Now, that is a fun idea, but will be a very large API-breaking change. This PR gives us a lot of the benefits TODAY, as opposed to waiting a long time for ALL of the benefits.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by john-science** [comment]
This little exercise has already born some fruit.

@keckler Notice the setting here `epsBurnTime` is in an "unknown" category? That's because this IS NOT an ARMI setting, but comes from an internal neutronics plugin. Now I want to remove those here, but will have to test if it breaks downstream unit tests.

@drewejohnson @albeanth When you organize settings like this, it makes it _really_ look like there should be a group of settings explicitly tied to the "axial expansion" plugin.

**Comment by albeanth** [comment]
> Now I want to remove those here, but will have to test if it breaks downstream unit tests.

I'd be in favor of removing them if it's not too big of a deal! 

> When you organize settings like this, it makes it really look like there should be a group of settings explicitly tied to the "axial expansion" plugin.

Yeah, that's not entirely surprising, tbh.

File: armi/tests/armiRun.yaml [hunk 1]
```diff
@@ -31,30 +51,23 @@ settings:
...
-  shuffleLogic: refSmallReactorShuffleLogic.py
-  smallRun: true
   summarizeAssemDesign: false
-  targetK: 1.002
-  verbosity: extra
-  versions:
-    armi: uncontrolled
+
+# unknown
+  epsBurnTime: 0.001
```
Comment (john-science) [inline armi/tests/armiRun.yaml:73]:
```suggestion
```

File: armi/tests/refTestCartesian.yaml [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
+# neutronics
+  epsFSAvg: 1e-06
+  epsFSPoint: 1e-06
+  loadPadElevation: 200.0
+
+# report
+  summarizeAssemDesign: false
+
+# unknown
+  epsBurnTime: 0.001
```
Comment (john-science) [inline armi/tests/refTestCartesian.yaml:30]:
```suggestion
```

File: armi/tests/tutorials/c5g7-settings.yaml [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
     armi: uncontrolled
+
+# neutronics
+  genXS: Neutron
+
+# report
+  genReports: false
+
+# unknown
+  fakeBad: 123456
```
Comment (albeanth) [inline armi/tests/tutorials/c5g7-settings.yaml:23]: Can we remove this?

**Review by albeanth — APPROVED** [review]
Just the one minor comment to try and remove that other unknown setting.

File: armi/tests/tutorials/c5g7-settings.yaml (hunk 3 above)
Comment (john-science) [reply armi/tests/tutorials/c5g7-settings.yaml:23]: Hmm... I mean, this is a unit tests file, and that is in the unit test.

I can look at re-writing that test.

Comment (john-science) [reply armi/tests/tutorials/c5g7-settings.yaml]: Okay, I changed the test. We're good.

File: armi/tests/test_lwrInputs.py [hunk 4]
```diff
@@ -63,7 +63,6 @@ def test_loadC5G7(self):
             streamVal = mock.getStdout()
             self.assertIn("UraniumOxide", streamVal, msg=streamVal)
             self.assertIn("SaturatedWater", streamVal, msg=streamVal)
-            self.assertIn("invalid settings: fakeBad", streamVal, msg=streamVal)
```
Comment (albeanth) [inline armi/tests/test_lwrInputs.py:66]: Ah I see, I didn't even think that it was there on purpose. This sort of behavior (checking that invalid settings are caught in settings validation) feels like it should be covered elsewhere, yeah? 

Just want to make sure we aren't losing something by ripping this out.

Comment (john-science) [reply armi/tests/test_lwrInputs.py:66]: We have this in several other places, not to worry.

**Pull request merged**